pkg/kv/kvserver/replica_learner_test.go

```diff
@@ -1533,3 +1533,276 @@ func TestMergeQueueSeesLearnerOrJointConfig(t *testing.T) {
 		require.False(t, desc.Replicas().InAtomicReplicationChange(), desc)
 	}
 }
+
+// Tests the accuracy of the 'range.snapshots.rebalancing.rcvd-bytes' and
+// 'range.snapshots.rebalancing.sent-bytes' metrics. This test adds a new
+// replica to a cluster, and during the process, a learner snapshot is sent to
+// the new replica.
+func TestRebalancingSnapshotMetrics(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	blockUntilSnapshotCh := make(chan struct{})
+	blockSnapshotsCh := make(chan struct{})
+	knobs, ltk := makeReplicationTestKnobs()
+	ltk.storeKnobs.ReceiveSnapshot = func(h *kvserverpb.SnapshotRequest_Header) error {
+		close(blockUntilSnapshotCh)
+		select {
+		case <-blockSnapshotsCh:
+		case <-time.After(10 * time.Second):
+			return errors.New(`test timed out`)
+		}
+		return nil
+	}
+	ctx := context.Background()
+	tc := testcluster.StartTestCluster(t, 2, base.TestClusterArgs{
+		ServerArgs:      base.TestServerArgs{Knobs: knobs},
+		ReplicationMode: base.ReplicationManual,
+	})
+	defer tc.Stopper().Stop(ctx)
+
+	scratchStartKey := tc.ScratchRange(t)
+
+	// Initial recording of metrics
+	senderRebalancingSentBytesBefore := getFirstStoreMetric(t, tc.Server(0), "range.snapshots.rebalancing.sent-bytes")
```

Comment:
nit: Since you repeat this logic a bit, you could perhaps create an helper func in like `getSnapshotCounters(serverID int)` which would return a struct 

```go
type testingSnapshotCounters {
     recoverySentBytes int64
     recoveryRcvdBytes int64
     ...
}
```

Given you use  this in both tests.

Then assert on the fields like you are doing below.